lib/rouge/lexer.rb

```diff
@@ -254,9 +254,12 @@ def mimetypes(*mts)
         (@mimetypes ||= []).concat(mts)
       end
 
+      VALID_ENCODINGS = %w(US-ASCII UTF-8 ASCII-8BIT).freeze
+      private_constant :VALID_ENCODINGS
+
       # @private
       def assert_utf8!(str)
-        return if %w(US-ASCII UTF-8 ASCII-8BIT).include? str.encoding.name
+        return if VALID_ENCODINGS.include? str.encoding.name
```

Comment:
Given this array isn't used anywhere else, I wonder if it would be preferable to memoise it. Something like:

```ruby
def assert_utf8!(str)
  @valid_encodings ||= %w(US-ASCII UTF-8 ASCII-8BIT)
  return if @valid_encodings.include? str.encoding.name
  raise EncodingError.new(
    "Bad encoding: #{str.encoding.names.join(',')}. " +
    "Please convert your string to UTF-8."
  )
end
```

What do you think? (I confess this is, like, 90% because Ruby constants always look gross to me :P)

Reply:
Since this is a class method, either approach would have the same result. (i.e. initialized as soon as `required`). The flipside is that constants can emit a warning if they get altered, ivars do not.

Reply:
Yeah, that does seem like a nice to have.